### PR TITLE
LLM: add qlora finetuning example using `trl.SFTTrainer`

### PR DESCRIPTION
**Pull request by plusbang**
## Description

Background: https://github.com/analytics-zoo/nano/issues/1047
First add example of providing `bigdl-llm` compatible `PeftModel` to `trl.SFTTrainer`

### 4. How to test?
- [x] Unit test
- [x] Local test

### CODE REVIEW

File: python/llm/example/GPU/LLM-Finetuning/QLoRA/trl-example/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Example of QLoRA Finetuning with BigDL-LLM
+
+This simple example demonstrates how to finetune a llama2-7b model use BigDL-LLM 4bit optimizations using [Intel GPUs](../../../README.md).
```
Comment (jason-dai) [inline python/llm/example/GPU/LLM-Finetuning/QLoRA/trl-example/README.md:3]: with TRL library on Intel GPU

Comment (plusbang) [reply python/llm/example/GPU/LLM-Finetuning/QLoRA/trl-example/README.md]: > with TRL library on Intel GPU

Have updated.

**Comment by jason-dai** [comment]
LGTM

**Review by jason-dai — APPROVED** [review]
LGTM

**Pull request merged**